apps/arena/lib/arena/game_updater.ex

```diff
@@ -598,4 +610,72 @@ defmodule Arena.GameUpdater do
       _ -> entity_id
     end
   end
+
+  defp spawn_power_ups(
+         %{game_state: game_state, game_config: game_config} = state,
+         victim,
+         amount
+       )
+       when amount > 0 do
+    last_id = game_state.last_id + 1
+
+    random_x = victim.position.x + Enum.random(-game_config.power_ups.distance_to_power_up..game_config.power_ups.distance_to_power_up)
+    random_y = victim.position.y + Enum.random(-game_config.power_ups.distance_to_power_up..game_config.power_ups.distance_to_power_up)
+
+    random_position = %{x: random_x, y: random_y}
+
+    power_up =
+      Entities.new_power_up(
+        last_id,
+        random_position,
+        victim.direction,
+        victim.id
+      )
+
+    put_in(state, [:game_state, :power_ups, last_id], power_up)
+    |> put_in([:game_state, :last_id], last_id)
+    |> spawn_power_ups(victim, amount - 1)
+  end
+
+  defp spawn_power_ups(state, _victim, _amount), do: state
+
+  defp get_of_power_ups(%{aditional_info: %{power_ups: power_ups}}, power_ups_per_kill) do
+    Enum.sort_by(power_ups_per_kill, fn %{amount: minimun} -> minimun end, :desc)
+    |> Enum.find(fn %{amount: minimun} ->
+      minimun <= power_ups
+    end)
+    |> case do
+      %{amount_of_drops: amount} -> amount
+      _ -> 0
+    end
+  end
+
+  defp handle_power_ups(players, power_ups) do
+    power_ups =
+      Map.reject(power_ups, fn {_power_up_id, power_up} ->
+        power_up.aditional_info.status == :TAKEN
+      end)
+
+    Enum.reduce(players, {players, power_ups}, fn {_player_id, player},
```

Comment:
If I followed this correctly seems like if multiple players are colliding with the same power up all of them will take

Reply:
You are totally right! if multiple player collide with a power up at the same time they'll both pick it up, i'll handle this adding a status check for the power up since this is being updated on each iteration